### DISCUSSION TITLE
Is there a way to automatically build a library, each time prior to running a specific test?

### DISCUSSION THREAD

**Discussion by krispin-gradient** [ThrowTheSwitch/Ceedling#952] [Q&A] [answered]
Is there a way to automatically build a library with `cmake`, prior to running a specific test, or group of tests?

In the CeedlingPacket docs, there is: `libraries:` "Add a collection of library paths to be included when linking", but (if I'm understanding right?) this is for libraries that are already built _before_ running a test.

[Ceedling/plugins/dependencies/readme.md](https://github.com/ThrowTheSwitch/Ceedling/blob/c778591acff5d089d142e6cb9e5fa4e0944cec6a/plugins/dependencies/README.md?plain=1#L17), points to the ability to make a library with tools such as cmake, but (if I'm understanding right?) for the purposes of _releasing_, rather than testing.

Can anyone recommend examples/resources/docs, re: how to fire off the cmake build of a library, in the context of building/running a specific test or group of tests?

**Comment by mvandervoord** [comment, marked as answer]
In the 0.3x releases, it wasn't possible. In the pre-release for 1.0, you want to look at the [dependencies](https://github.com/ThrowTheSwitch/Ceedling/blob/test/ceedling_0_32_rc/plugins/dependencies/README.md) plugin.

**Comment by krispin-gradient** [comment]
Thanks!
